bin/transform-genbank

```diff
@@ -0,0 +1,317 @@
+
+    return genbank_data
+
+
+def generate_hardcoded_metadata(genbank_data: pd.DataFrame) -> pd.DataFrame:
+    """
+    Returns a DataFrame with a column for GenBank accession plus
+    additional columns containing hardcoded metadata.
+    """
+    hardcoded_metadata = pd.DataFrame(genbank_data['genbank_accession'])
+    hardcoded_metadata['virus']             = 'ncov'
+    hardcoded_metadata['gisaid_epi_isl']    = '?'
+    hardcoded_metadata['segment']           = 'genome'
+    hardcoded_metadata['age']               = '?'
+    hardcoded_metadata['sex']               = '?'
+    hardcoded_metadata['originating_lab']   = '?'
+    hardcoded_metadata['submitting_lab']    = '?'
+
+    hardcoded_metadata['url'] = hardcoded_metadata['genbank_accession'] \
+        .apply(lambda x: f"https://www.ncbi.nlm.nih.gov/nuccore/{x}")
+
+    return hardcoded_metadata
+
+
+def update_metadata(genbank_data: pd.DataFrame) -> pd.DataFrame:
+    """
+    Update metadata with hardcoded metadata and update with curated
+    annotations if provided. Returns the modified DataFrame.
+    """
+    hardcoded_metadata = generate_hardcoded_metadata(genbank_data)
+    genbank_data = genbank_data.merge(hardcoded_metadata)
+
+    if args.annotations:
+        # Use the curated annotations tsv to update any column values
+        user_provided_annotations = pd.read_csv(args.annotations, header=None, sep='\t', comment="#")
+        for index, (genbank_accession, key, value) in user_provided_annotations.iterrows():
+            # Strip any whitespace remaining from inline comments after the final column.
+            if isinstance(value, str):
+                value = value.rstrip()
+            genbank_data.loc[genbank_data['genbank_accession'] == genbank_accession, key] = value
+
+    # if division is blank, replace with country data, to avoid unexpected effects when subsampling by division
+    # (where an empty division is counted as a 'division' group)
+    genbank_data.loc[pd.isnull(genbank_data['division']), 'division'] = genbank_data['country']
+
+    # Set `region_exposure` equal to `region` if it wasn't added by annotations
+    if 'region_exposure' in genbank_data:
+        genbank_data['region_exposure'].fillna(genbank_data['region'], inplace=True)
+    else:
+        genbank_data['region_exposure'] = genbank_data['region']
+
+    # Set `country_exposure` equal to `country` if it wasn't added by annotations
+    if 'country_exposure' in genbank_data:
+        genbank_data['country_exposure'].fillna(genbank_data['country'], inplace=True)
+    else:
+        genbank_data['country_exposure'] = genbank_data['country']
+
+    # Set `division_exposure` equal to `division` if it wasn't added by annotations
+    if 'division_exposure' in genbank_data:
+        genbank_data['division_exposure'].fillna(genbank_data['division'], inplace=True)
+    else:
+        genbank_data['division_exposure'] = genbank_data['division']
```

Comment:
Same comment re: redundancy in the first half of this function and `transform-gisaid`'s `update_metadata()` as well